rotors_gazebo_plugins/src/gazebo_motor_model.cpp

```diff
@@ -252,85 +362,108 @@ void GazeboMotorModel::WindSpeedCallback(GzWindSpeedMsgPtr& wind_speed_msg) {
 }
 
 void GazeboMotorModel::UpdateForcesAndMoments() {
-  motor_rot_vel_ = joint_->GetVelocity(0);
-  if (motor_rot_vel_ / (2 * M_PI) > 1 / (2 * sampling_time_)) {
-    gzerr << "Aliasing on motor [" << motor_number_
-          << "] might occur. Consider making smaller simulation time steps or "
-             "raising the rotor_velocity_slowdown_sim_ param.\n";
-  }
-  double real_motor_velocity = motor_rot_vel_ * rotor_velocity_slowdown_sim_;
-  // Get the direction of the rotor rotation.
-  int real_motor_velocity_sign = (real_motor_velocity > 0) - (real_motor_velocity < 0);
-  //Assuming symmetric propellers (or rotors) for the force calculation.
-  double force = turning_direction_ * real_motor_velocity_sign * real_motor_velocity * real_motor_velocity * motor_constant_;
+
+  switch (motor_type_) {
+  case (motor_type::POSITION): {
+    double err = joint_->GetAngle(0).Radian() - ref_motor_input_;
+    double force = pids_.Update(err, sampling_time_);
+    joint_->SetForce(0, force);
+  } break;
+
+  case (motor_type::FORCE):
+    joint_->SetForce(0, std::min(ref_motor_input_, max_force_));
+    break;
+
+  default: // motor_type::VELOCITY
+  {
```

Comment:
fix style, maybe run clang-format over the changed lines :)

Reply:
There seem to be different clang format options for switch cases, which do you go with?

Reply:
nevermind you mention it later. Thanks!